src/lib/offline-api/validator/field-groups-count.ts

```diff
@@ -0,0 +1,34 @@
+import { ApiHook } from '..'
+import errorMessages from './errors'
+import { EditorInterfacePayloadValidator } from './editor-interface'
+import { EditorInterfaces } from '../../entities/content-type'
+import { collectFieldGroupIds } from '../../utils/editor-layout'
+import { PayloadValidationError } from '../../interfaces/errors'
+
+export default class FieldGroupsCountValidator implements EditorInterfacePayloadValidator {
+  public hooks = [ApiHook.SaveEditorInterface]
+
+  public validate (editorInterface: EditorInterfaces): PayloadValidationError[] {
+    const errors: PayloadValidationError[] = []
+    const editorLayout = editorInterface.getEditorLayout()
+
+    const groupsCount = collectFieldGroupIds(editorLayout).length
+    const tabsCount = editorLayout.length
+
+    if (groupsCount < 2) {
+      errors.push({
+        type: 'InvalidPayload',
+        message: errorMessages.editorLayout.TOO_FEW_FIELD_GROUPS()
+      })
+    }
+
+    if (groupsCount - tabsCount > 15) {
```

Comment:
Could we have these limits as outer constants and use them also in the related error messages?

Reply:
Saved them as constants. They're still hardcoded in the readme files, so I left them in the tests as well.